### PR TITLE
[Layout] Rename ASLayoutable to ASLayoutElement

### PR DESCRIPTION
**Pull request by hannahmbanana**
Rename for clarity:

- `ASLayoutable` ---> `ASLayoutElement`

### CODE REVIEW

**Comment by hannahmbanana** [comment]
@maicki: ready for review on Monday

File: AsyncDisplayKit/Layout/ASLayout.h [hunk 1]
```diff
@@ -81,20 +81,20 @@ ASDISPLAYNODE_EXTERN_C_END
...
  * Convenience class initializer for layout construction.
  *
- * @param layoutable The backing ASLayoutable object.
+ * @param layoutElement The backing ASLayoutElement object.
  * @param size             The size of this layout.
  * @param position         The position of this layout within its parent (if available).
  * @param sublayouts       Sublayouts belong to the new layout.
  */
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.h]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.h [hunk 2]
```diff
@@ -105,11 +105,11 @@ ASDISPLAYNODE_EXTERN_C_END
  * or for ASLayoutSpec subclasses that are referencing the "self" level in the layout tree,
  * or for creating a sublayout of which the position is yet to be determined.
  *
- * @param layoutable  The backing ASLayoutable object.
+ * @param layoutElement  The backing ASLayoutElement object.
  * @param size              The size of this layout.
  * @param sublayouts        Sublayouts belong to the new layout.
  */
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.h]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.h [hunk 3]
```diff
@@ -118,10 +118,10 @@ ASDISPLAYNODE_EXTERN_C_END
  * Best used for creating a layout that has no sublayouts, and is either a root one
  * or a sublayout of which the position is yet to be determined.
  *
- * @param layoutable The backing ASLayoutable object.
+ * @param layoutElement The backing ASLayoutElement object.
  * @param size             The size of this layout.
  */
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.h]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 4]
```diff
@@ -52,12 +52,12 @@ @implementation ASLayout
 
 @dynamic frame, type;
 
-- (instancetype)initWithLayoutable:(id<ASLayoutable>)layoutable
+- (instancetype)initWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 5]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
 
 #pragma mark - Class Constructors
 
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 6]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
                                 size:(CGSize)size
                             position:(CGPoint)position
                           sublayouts:(nullable NSArray<ASLayout *> *)sublayouts
 {
-  return [[self alloc] initWithLayoutable:layoutable
+  return [[self alloc] initWithLayoutElement:layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 7]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
                                      size:size
                                  position:position
                                sublayouts:sublayouts];
 }
 
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 8]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
                                 size:(CGSize)size
                           sublayouts:(nullable NSArray<ASLayout *> *)sublayouts
 {
-  return [self layoutWithLayoutable:layoutable
+  return [self layoutWithLayoutElement:layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 9]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
                                size:size
                            position:CGPointNull
                          sublayouts:sublayouts];
 }
 
-+ (instancetype)layoutWithLayoutable:(id<ASLayoutable>)layoutable size:(CGSize)size
++ (instancetype)layoutWithLayoutElement:(id<ASLayoutElement>)layoutElement size:(CGSize)size
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 10]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
 {
-  return [self layoutWithLayoutable:layoutable
+  return [self layoutWithLayoutElement:layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

File: AsyncDisplayKit/Layout/ASLayout.mm [hunk 11]
```diff
@@ -97,38 +97,38 @@ - (instancetype)init
...
                                size:size
                            position:CGPointNull
                          sublayouts:nil];
 }
 
 + (instancetype)layoutWithLayout:(ASLayout *)layout position:(CGPoint)position
 {
-  return [self layoutWithLayoutable:layout.layoutable
+  return [self layoutWithLayoutElement:layout.layoutElement
```
Comment (maicki) [inline AsyncDisplayKit/Layout/ASLayout.mm]: Adjust indentation

**Comment by hannahmbanana** [comment]
@maicki: addressed your comments

**Comment by maicki** [comment]
LGTM

**Comment by hannahmbanana** [comment]
@maicki: This one had 15 rebase conflicts, so another quick review might be safe.

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 12]
```diff
@@ -299,8 +299,8 @@ - (void)_initializeInstance
   _contentsScaleForDisplay = ASScreenScale();
   _displaySentinel = [[ASSentinel alloc] init];
   
-  _style = [[ASLayoutableStyle alloc] init];
-  _size = ASLayoutableSizeMake();
```
Comment (hannahmbanana) [inline AsyncDisplayKit/ASDisplayNode.mm]: this _size struct was left behind and is now contained by the _style object

**Comment by appleguy** [comment]
Reviewed - dang, a long one - and built this locally as well (unit tests, examples, and framework target).

It looks like there was a lot of rebase thrash on this one, so I'm going to merge it now while it's clean.  It will probably impact a few other diffs, but rebasing them would be much easier than this one.

**Pull request merged**